### PR TITLE
move traits to separte package

### PR DESCRIPTION
**Pull request by xoviat**

### CODE REVIEW

File: embassy/src/lib.rs [hunk 1]
```diff
@@ -9,10 +9,9 @@
...
 pub mod executor;
-pub mod flash;
-pub mod gpio;
 pub mod interrupt;
 pub mod io;
 pub mod time;
-pub mod uart;
 pub mod util;
+
+pub use embasy_traits::{flash, gpio, uart};
```
Comment (Dirbaio) [inline embassy/src/lib.rs:17]: Maybe it's better to `pub use embassy_traits as traits`, so they're grouped?

Comment (Dirbaio) [reply embassy/src/lib.rs:17]: Also the io traits should probably go there too?

**Comment by xoviat** [comment]
@Dirbaio I cannot solve the defmt issue.

**Comment by Dirbaio** [comment]
Done, and squashed it for you.

Thanks for the PR :)

**Pull request merged**